### PR TITLE
Templated queries

### LINKED ISSUE #52: Support for templated queries
for example, a query that gives all attribute information about a specific manufacturer --> be able to select that manufacturer from a dropdown

### PR DESCRIPTION
**Pull request by axellorreyne**
Fixes #52

### CODE REVIEW

File: src/components/ListResultTable/ListResultTable.jsx [hunk 1]
```diff
@@ -1,59 +1,83 @@
...
+
+            for (const [variableName, variableValue] of Object.entries(variables)) {
+                queryParams.set(variableName, variableValue)
+            }
+
+            navigate(
+                `?${queryParams.toString()}`
+            );
+        }
+    }, [location.search, navigate]);
```
Comment (axellorreyne) [inline src/components/ListResultTable/ListResultTable.jsx:41]: The history gets filled up with multiple instances of the exact same URL when loading a templated query with certain values for variables, so trying to go back in the browser's history results in returning to the same page for the templated query with the same values for variables multiple times.

**Comment by mvanbrab** [comment]
TODO: Document templated queries in README.md
(a) in the configuration file structure
(b) description

**Comment by mvanbrab** [comment]
The UI experience is rather unexpected:
- once a value from the drop down list is chosen, one has to push the save button
- then the drop down list disappears and the query is executed.
Now it is not possible to redo the query with a different choice from the drop down list ???
Unless... the user switches to a different query and comes back.

It would be more intuitive if the drop down list stay visible all the time and the query is executed at every change of the value(s)...

Note: to be demonstrated and tested for the case of more than one drop down list.

**Comment by mvanbrab** [comment]
At this moment, templates only work for string constants. This should be extended to anything (including URLs and primitive values).
Proposal: do not surround values with double quotes; expect the user to give literal input.
Example test input (not a dictatical example, I know):
Modified query:
```
PREFIX schema: <http://schema.org/>
  
SELECT ?name WHERE {
  ?list schema:name ?listTitle;
    schema:name ?name;
    schema:genre $genre;
    schema:sameAs ?$sameAsUrl;
}
```
Modified configuration (variables only):
```
      "variables": {
        "genre": ["\"Romantic\"", "\"Baroque\"", "\"Classical\""],
        "sameAsUrl": ["<https://en.wikipedia.org/wiki/Ludwig_van_Beethoven>", "<https://en.wikipedia.org/wiki/Wolfgang_Amadeus_Mozart>"]
      },
```

**Comment by mvanbrab** [comment]
Rendering of the drop down list:
- the label should not be converted:
  - "genre" should be displayed as "genre" (not "Genre")
  - "genreCamelCaseTest" should be displayed as "genreCamelCaseTest" (not "Genre camel case test")
- there should be same padding between the label and the chosen value afbeelding

**Comment by mvanbrab** [comment]
Query templates should be covered in the tests.

File: src/components/ListResultTable/ListResultTable.jsx (hunk 1 above)
Comment (mvanbrab) [reply src/components/ListResultTable/ListResultTable.jsx:41]: Better put this as comments in the code...

Comment (mvanbrab) [reply src/components/ListResultTable/ListResultTable.jsx:41]: Looking at the browser history in Firefox, it appears that there are only 2 entries for the last visited templated query: one without parameters in the URL (last but one entry) and one including parameters (last entry). When clicking the back arrow, it appears that the browser goes back to the last but one entry, but then immediately jumps forward again to the last entry, as if the above code is executed again.
See also my comment below about the UI experience - maybe to be solved all together.

Comment (mvanbrab) [reply src/components/ListResultTable/ListResultTable.jsx]: **Solved in 151b25b44aef96c994be9881d770e3fc2459419f**.

**Pull request merged**